### ISSUE TITLE
Requirement Documents

### ISSUE THREAD

**Issue by timolinn** [Sedronik/remote-doctor#1]
@cyberdroidmann I suggest we create a detailed requirements document for `remote doctor`. The document should be for both technical and non-technical readers. Check the list below, they are the things I think we should discuss extensively and put them into writing.

- Problem we are solving (with real world stats if possible).
- Goal of the project.
- Supported/Preferred platforms with reasons for choice.
- Technologies/Tools to use and why.
- Architecture and System design.

There's more, but those are what I can think of right now. I'm not sure we need to be exhaustive about the subjects.

Your thoughts?

**Comment by bitsofsteve** [comment]
## Background Story

**The 24th Annual Scientific Conference of the National Association of Community Health Practitioners of Nigeria (NACHPN), that held in Katsina State, Nigeria on April 16 - 20th, 2018. One of the major concerns that was discussed in the  conference was "Universal Healthcare Coverage", by 'Universal' they mean, 'Nationwide'.**

The problem with that first paragrapgh is that in reality, only about 10% of clinics in village hospitals here in Nigeria actually witness daily/weekly doctors visit, the remaining 90% of rural clinics might witness a doctor's visit once every 2 weeks, or once a month. This is in no way supports the idea of "Global Health Coverage". 


### Extrapolated reasons for this problem

*  High Urbanization of doctors
*  Unwillingness by doctors to travel deep into villages to adminster basic care
*  Low number of doctors 
*  High level of overseas migration by doctors


### Effects of this problem

*  Serious decline in health and wellness of people in rural areas
*  Increase in mortality rate
*  Decline in patient education
* High rise of infectious diseases
* Increase in self medication and treatment
* Unprofessionals/Quacks trying to cover the gap



# Goal of the Project

### Mission

To bridge the gap between urbanized & abroad healthcare professionals in Nigeria, using an open sourced, state-of-the-arts cloud-based telehealth care technology.

### Vision
A world where the children, women, men, elderly and the rank and file of the society have access to healthcare professionals, and state-of-the-art care.

**Comment by timolinn** [comment]
Good job @cyberdroidmann 

I remember we talked about Network access for rural areas, which I believe is mostly poor. Glo's 4G lte might be the cost efficient way to obtain network access. This also brings into light that the application has to be able to work well in as low internet bandwidth as possible.

I was looking we build out the application with Nodejs, I'm not sure if this the best option now, because most Nodejs apps are known to be memory intensive and can be prone to memory leaks because of Node's event driven architecture.

**Comment by bitsofsteve** [comment]
> Good job @cyberdroidmann
> 
> I remember we talked about Network access for rural areas, which I believe is mostly poor. Glo's 4G lte might be the cost efficient way to obtain network access. This also brings into light that the application has to be able to work well in as low internet bandwidth as possible.
> 
> I was looking we build out the application with Nodejs, I'm not sure if this the best option now, because most Nodejs apps are known to be memory intensive and can be prone to memory leaks because of Node's event driven architecture.

Network compatibility is a huge consideration, thanks for reiterating.


If we are not gonna build it out in NodeJs, the next considerable option is Flask, oh well we need to do alot of research about tradeoffs

**Comment by timolinn** [comment]
> If we are not gonna build it out in NodeJs, the next considerable option is Flask, oh well we need to do alot of research about tradeoffs

Yep! tradeoffs! For the sake of real time scenarios and concurrency, I think Nodejs is ahead of Python. Another option is Golang or Rust. Would probably be a good idea to use node for this particular version since it's just a proof of concept and not the actual production application.
